### PR TITLE
Base of the cross-distro installer, some fixes

### PR DESCRIPTION
**Pull request by sTiKyt**
Fix: Removed some weird characters and slightly improved README.md

Add: Installer asks for root if not provided instead of exiting it

Fix: Installer can determine which package manager it's running with and doesn't have to rely on specific distro

Fix: Installer won't let steps that require "apt" run and will return user to the selection menu, if host distro uses something else (this is a temporary fool-proof solution until the installation steps themselves can be improved to work on different distros)

Fix: Forced users to clearly agree to using default credentials if they don't provide API_ID/HASH, so they don't get banned by mistake and are well aware of the risks

Fix: Made some prompts more clear and easier to understand (At least the ones I was confused with and had to clarify about in chat)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,36 +128,39 @@ If you're using Docker Compose version 2.x, use the following commands to start
...
-
 ### Install Git
 
 > [!TIP]
-> Ignore if already installed
-
-```shell
-sudo apt install git
-```
+> Ensure you have the git installed
```
Comment (AbhiTheModder) [inline README.md:139]: Would be better to add different OS'es installation commands here.

File: install.sh [hunk 2]
```diff
@@ -1,24 +1,58 @@
...
+    ;;
+  pacman)
+    pacman -S --noconfirm python python-virtualenv git wget || exit 2
+    ;;
+esac
 
-if [[ -d "Moon-Userbot" ]]; then
-  cd Moon-Userbot
-elif [[ -f ".env.dist" ]] && [[ -f "main.py" ]] && [[ -d "modules" ]]; then
-  :
```
Comment (AbhiTheModder) [inline install.sh:21]: The additions were made to confirm that the directory is not merely a placeholder `Moon-Userbot`. Please implement an equivalent or improved conditional.

File: install.sh [hunk 3]
```diff
@@ -28,33 +62,53 @@ if [[ -f ".env" ]] && [[ -f "my_account.session" ]]; then
   echo "It seems that Moon-Userbot is already installed. Exiting..."
   exit
 fi
+# Create a virtual environment inside the cloned repository and activate it
+python3 -m venv venv
+source venv/bin/activate
```
Comment (AbhiTheModder) [inline install.sh:67]: To enhance the setup process, begin by inquiring whether the user wishes to configure the virtual environment before proceeding.

File: install.sh [hunk 4]
```diff
@@ -28,33 +62,53 @@ if [[ -f ".env" ]] && [[ -f "my_account.session" ]]; then
...
+echo "Leave empty to use defaults (please note that using default keys is a **very bad idea** and significantly increases your ban chances)"
 read -r -p "API_ID > " api_id
 
+# Default API_ID and API_HASH
 if [[ $api_id = "" ]]; then
-  api_id="2040"
-  api_hash="b18441a1ff607e10a989891a5462e627"
+  echo "You have chosen to use the default API_ID and API_HASH, which is strongly discouraged."
+  echo "Please type 'I agree' to confirm that you understand the risks and still wish to proceed."
+  read -r -p "Confirmation > " confirmation
```
Comment (AbhiTheModder) [inline install.sh:85]: This is nice; I liked it.

File: install.sh [hunk 5]
```diff
@@ -28,33 +62,53 @@ if [[ -f ".env" ]] && [[ -f "my_account.session" ]]; then
...
+  else
+    echo "Confirmation not provided. Exiting..."
+    exit 1
+  fi
 else
   read -r -p "API_HASH > " api_hash
 fi
-
+# Prompt for PM PERMIT warn limit
+# PM PERMIT warn limit is the number of messages a user can receive from others before giving them a warning
```
Comment (AbhiTheModder) [inline install.sh:97]: It would be better to also mention that the `antipm` plugin needs to be enabled for it to work.

File: install.sh [hunk 6]
```diff
@@ -28,33 +62,53 @@ if [[ -f ".env" ]] && [[ -f "my_account.session" ]]; then
...
 echo
 echo "SET PM PERMIT warn limit"
+#Now below is more clear version:
```
Comment (AbhiTheModder) [inline install.sh:100]: space :>

File: install.sh [hunk 7]
```diff
@@ -28,33 +62,53 @@ if [[ -f ".env" ]] && [[ -f "my_account.session" ]]; then
...
+echo "Enter the number of messages others can send you before receiving a warning, and eventually a ban or leave empty for default (3)"
```
Comment (AbhiTheModder) [inline install.sh:101]: same here, would be better to also mention that the antipm plugin needs to be enabled for it to work.

File: install.sh [hunk 8]
```diff
@@ -125,47 +179,64 @@ echo "Learn How to Get One --> https://github.com/VisionCraft-org/VisionCraft?ta
...
+        systemctl enable mongodb
+      fi
+      systemctl start mongodb
+
+      db_url=mongodb://localhost:27017
+      db_name=Moon_Userbot
+      db_type=mongodb
+      break
+      ;;
+    *)
```
Comment (AbhiTheModder) [inline install.sh:228]:
```suggestion
    3)
```

File: install.sh [hunk 9]
```diff
@@ -187,40 +258,53 @@ SECOND_SESSION=${second_session}
...
-    if ! command -v pm2; then
-      curl -fsSL https://deb.nodesource.com/setup_17.x | bash
-      apt install nodejs -y
-      npm install pm2 -g
-      su -c "pm2 startup" $SUDO_USER
-      env PATH=$PATH:/usr/bin /usr/lib/node_modules/pm2/bin/pm2 startup systemd -u $SUDO_USER --hp /home/$SUDO_USER
-    fi
-    su -c "pm2 start main.py --name Moon --interpreter python3" $SUDO_USER
-    su -c "pm2 save" $SUDO_USER
+su -c "python3 install.py ${install_type}" $SUDO_USER || exit 3
```
Comment (AbhiTheModder) [inline install.sh:266]: The `install_type` is supposed to be set within the `while` loop based on user input, right?

**Review by AbhiTheModder — CHANGES_REQUESTED** [review]

**Comment by AbhiTheModder** [comment]
@sTiKyt, thank you for the pull request; it's greatly appreciated. Please address the feedback before merging.

File: install.sh (hunk 6 above)
Comment (AbhiTheModder) [inline install.sh:100]:
```suggestion
# Now below is more clear version:
```

File: install.sh (hunk 5 above)
Comment (AbhiTheModder) [inline install.sh:97]:
```suggestion
# PM PERMIT warn limit is the number of messages a user can receive from others before giving them a warning, requires `antipm` plugin to be enabled
```

File: install.sh (hunk 2 above)
Comment (sTiKyt) [reply install.sh:21]: Fixed in 76ecbb925110793255cecdbb9aa0daefe83829b9

File: install.sh (hunk 3 above)
Comment (sTiKyt) [reply install.sh]: Adressed in 257db174475db0708c1daabb1926752edc9fe95b

File: install.sh (hunk 8 above)
Comment (sTiKyt) [reply install.sh]: Fixed in 722cf0cd76e8bd0374cf3cc456c8b7106159c09a

File: install.sh (hunk 9 above)
Comment (sTiKyt) [reply install.sh:273]: Not sure what you mean, and also not sure what even this "install.py" file for

Comment (AbhiTheModder) [reply install.sh:273]: > Not sure what you mean, and also not sure what even this "install.py" file for, it seems a bit useless/replaceable with bash in the main file?

variable `install_type` is being used even before it's assigned.

Comment (sTiKyt) [reply install.sh]: Fixed in b891226993c1a6620d589965226fc817a9cdf52e

**Review by AbhiTheModder — APPROVED** [review]

**Pull request merged**